### PR TITLE
17wf/urls

### PR DESCRIPTION
**Pull request by Piwakk**

### CODE REVIEW

File: associations/permissions/marketplace.py [hunk 1]
```diff
@@ -72,8 +72,8 @@ def get_role_in_marketplace(request):
 
 class MarketplacePermission(BasePermission):
     """
-               | Enabled | Disabled |
-        Admin  | CRUD    | CRUD     |
+               | Enabled | Disabled |\n
```
Comment (abocquet) [inline associations/permissions/marketplace.py]: trailing \n

File: associations/tests/marketplace/test_balance.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def test_if_logged_in_then_can_retrieve_balances(self):
 
     def test_if_library_administrator_then_can_retrieve_all_users_balances(self):
         self.login('17market_biero')
-        res = self.get('marketplace/balance/biero/')
+        res = self.get('marketplace/biero/balance/')
```
Comment (abocquet) [inline associations/tests/marketplace/test_balance.py]: 👍

File: authentication/permissions.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        Admin  | RU          | RU           |
+        Simple | RU          | R            |
+
+        The fields which can be updated are specified in the UserSerializer.
+    """
+
+    message = 'You are not allowed to edit this profile.'
+
+    def has_permission(self, request, view):
+        if request.method in ('POST', 'DELETE',):
```
Comment (abocquet) [inline authentication/permissions.py]: `return request.method not in ('POST', 'DELETE',)` ?

File: authentication/permissions.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return True
+
+    def has_object_permission(self, request, view, target_user):
+        if request.user.is_admin:
+            return True
+        else:
+            if request.method in SAFE_METHODS:
+                return True
+            elif request.method in ('PATCH', ):
+                return target_user.id == request.user.id
```
Comment (abocquet) [inline authentication/permissions.py]: `return request.user.is_admin or request.method in SAFE_METHODS or (request.method in ('PATCH', ) and target_user.id == request.user.id)` ?
Ok maybe a bit long for a oneliner, but there may be no need for nested `if`/ `else`

File: authentication/serializers/user.py [hunk 5]
```diff
@@ -7,16 +7,16 @@
...
-                  "nickname", "birthday", "phone",
-                  "room", "address", "city_of_origin",
-                  "option", "is_ast", "is_isupfere", "is_in_gapyear",
-                  "sports", "roommate", "minesparent",
-                  "is_1A", 'promo'
-                  )
+        read_only_fields = ('first_name', 'last_name', 'birthday', 'email', 'year_of_entry', 'student_type', 'is_active',
+                            'is_admin',)
+        fields = read_only_fields + ('nickname', 'phone', 'room', 'address', 'city_of_origin', 'option', 'sports',
+                                     'roommate', 'minesparent',)
```
Comment (abocquet) [inline authentication/serializers/user.py]: Clever

File: authentication/views/profile.py [hunk 6]
```diff
@@ -1,50 +1,73 @@
...
-        if self.action == 'list' and self.request.query_params.get('startswith') is not None:
-            return UserShortSerializer
-        else:
-            return UserSerializer
+        return super(ProfileViewSet, self).get_object()
+
+    def update(self, request, *args, **kwargs):
+        """
+            Can X update Y? \n
+            X / Y       | Herself  | Another user | \n
```
Comment (abocquet) [inline authentication/views/profile.py]: `themself`

File: forum/urls.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+
+router = BulkRouter()
+
+router.register(r'', ThemeViewSet)
+router.register(r'theme', TopicViewSet)
+router.register(r'topic', MessageForumViewSet)
+
+urlpatterns = [
+    path('message-forum-vote/', NewVoteMessageView.as_view(), name='new-vote-message'),
```
Comment (abocquet) [inline forum/urls.py]: no need for `forum` in the url since they are nested now

File: authentication/permissions.py (hunk 4 above)
Comment (Piwakk) [reply authentication/permissions.py]:
```python
if request.user.is_admin:
    return True
else:
    return request.method in SAFE_METHODS or \
           (request.method in ('PATCH', ) and target_user.id == request.user.id)
```

File: authentication/views/profile.py (hunk 6 above)
Comment (Piwakk) [reply authentication/views/profile.py]: `themselves`*

**Pull request merged**